include/rabbitmq-c/amqp.h

```diff
@@ -2433,6 +2433,26 @@ AMQP_EXPORT
 int AMQP_CALL amqp_set_rpc_timeout(amqp_connection_state_t state,
                                    const struct timeval *timeout);
 
+
+/**
+ * amqp_publisher_confirm_wait
+ *
+ * Wait for a publisher confirm when the connection is in select mode.
+ * If the response has a `reply_type` of `AMQP_RESPONSE_LIBRARY_EXCEPTION` _and_
+ * the `library_error` is `AMQP_STATUS_UNEXPECTED_STATE`, then the frame
+ * received was likely not an ack.
+ *
+ * \param [in] state connection state
+ * \param [out] pointer to where the ACK details should go
+ * \param [in] timeout when waiting for the frame. Passing NULL will result in
+ * blocking behavior
+ * \returns amqp_rpc_reply_t *
+ */
+AMQP_EXPORT
+amqp_rpc_reply_t AMQP_CALL amqp_publisher_confirm_wait(
+    amqp_connection_state_t state, amqp_envelope_t *envelope,
```

Comment:
Could you describe the intended purpose of the envelope parameter?

Seems like this may not be required.

Reply:
Yeah, it's the only thing that'll tell you which channel the ACK came through. The code for the `amqp_basic_ack_t` is correct per spec, naturally, but it can be helpful to know which channel the ACK came on.

Reply:
Did you have any more questions or requests for change on this?

Reply:
If all you need is the channel id, then use `amqp_channel_t*` as an output parameter.  

Alternatively consider using an `amqp_channel_t` as an input parameter, then only wait for the confirm on this channel.

Reply:
I like the pulling the channel as an output parameter, and I'll combine that change with the method out param as well. I think making it filter by channel ID may be outside the scope of what I'm doing here, and the channel out param will allow the caller to easily implement that if they want to.